src/store/rewards/saga.ts

```diff
@@ -68,19 +68,30 @@ export function* syncRewardsAndTokenPrice() {
   if (!featureFlags.enableRewards) {
     return;
   }
-
-  yield spawn(syncFetchRewards);
   yield spawn(syncMEOWPrice);
+  yield spawn(syncFetchRewards);
```

Comment:
Just as an FYI, since these are both spawned to run "in the background", it doesn't matter which order they're in.

Reply:
got it. I actually wanted to do something like
```
yield call(syncMEOWPrice);
yield spawn(syncFetchRewards);
```

So that when we're doing `syncFetchRewards`, we have the actual `MEOW` token price in state. (so that when we're calculating the total rewards in the UI, it's not 0 for the first time).

But .. it didn't work, so i added the check in `checkNewRewardsLoaded` (to fetch the token price first before showing the rewards)